app/src/main/java/com/github/se/cyrcle/MainActivity.kt

```diff
@@ -59,7 +67,8 @@ class MainActivity : ComponentActivity() {
               reviewViewModel,
               userViewModel,
               mapViewModel,
-              addressViewModel)
+              addressViewModel,
+              PermissionsManager.areLocationPermissionsGranted(this))
```

Comment:
I believe this is why you need to relaunch the app for the permission to work. This runs as soon as you create the activity aka launch the app. So you haven't had the chances to accept permission. 

The solution should be to delay this line of code

Reply:
That's a very insightful comment I will make sure to test your hypothesis next sprint !